simple_app/utils.py

```diff
@@ -204,53 +316,96 @@ def __init__(self, *args, **kwargs):
         return
 
     def validate_sqlfield(self, field):
+        """
+        Validating SQL queries before they can be submitted
+        Parameters
+        ----------
+        field
+            The data within the query form
+        """
+        forbidden = ('update', 'drop', 'truncate', 'grant', 'commit', 'create', 'replace', 'alter', 'insert')
         db = SimpleDB(self.db_file, connection_arguments={'check_same_thread': False})  # open database
+
+        # check query field has data within
         if (query := field.data) is None or query.strip() == '':  # content in main searchbar
             raise ValidationError('Empty field')
+
+        # check for anything which would be problematic for sql to deal with
         try:
-            querylow: str = query.lower()
-            if not querylow.startswith('select') or 'from' not in querylow or \
-                    ('join' in querylow and not any([substr in querylow for substr in ('using', 'on')])):
-                raise BadSQLError('Queries must start with "select" and contain "from".'
-                                  ' Also, if performing a join, you must provide either "using" or "on".')
+            query_low: str = query.lower()
+
+            # select and from required
+            if not query_low.startswith('select') or 'from' not in query_low:
+                raise BadSQLError('Queries must start with "select" and contain "from".')
+
+            # checking joins are done correctly
+            if 'join' in query_low and not any([sub_string in query_low for sub_string in ('using', 'on')]):
+                raise BadSQLError('When performing a join, you must provide either "using" or "on".')
```

Comment:
I believe there are implicit joins that can be performed, though I think they don't use `join` at all. For example, doing something like `select * from Sources,Photometry`, though it's likely things like that don't work with sqlite (trying it gave me a 500 error)- I don't think we need to account for it, though.